pkg/planner/core/plan_cache.go

```diff
@@ -516,6 +517,10 @@ func rebuildRange(p Plan) error {
 		}
 		return nil
 	case *BatchPointGetPlan:
+		if x.TblInfo.GetPartitionInfo() != nil && fixcontrol.GetBoolWithDefault(sctx.GetSessionVars().OptimizerFixControl, fixcontrol.Fix33031, false) {
+			sctx.GetSessionVars().StmtCtx.UseCache = false
```

Comment:
Should use `SetSkipPlanCache`

Reply:
I will keep `UseCache = false`, since `SetSkipPlanCache` will also append a warning, which is already done if `rebuildRange` fails, like here. So to avoid duplicate warnings, I will keep this as-is.

Reply:
We agreed to remove the `UseCache = false` and only return an error, and accept possible duplicate warnings.